src/code/stores/graph-store.coffee

```diff
@@ -146,9 +146,31 @@ GraphStore  = Reflux.createStore
     @_graphUpdated()
     @updateListeners()
 
+  isUniqueTitle: (title, skipNode, nodes=@getNodes()) ->
+    nonUniqueNode = (otherNode) ->
+      sameTitle = otherNode.title is title
+      if skipNode then sameTitle and otherNode isnt skipNode else sameTitle
+    not _.find nodes, nonUniqueNode
+
+  ensureUniqueTitle: (node, newTitle=node.title) ->
+    nodes = @getNodes()
+    defaultTitle = tr "~NODE.UNTITLED"
+    if newTitle is defaultTitle or not @isUniqueTitle newTitle, node, nodes
```

Comment:
I was playing around with this before merging it just now and the first node created gets a default name of "Untitled 2", which seems counter-intuitive. My inclination would be to remove the `defaultTitle` check altogether. If the user really wants to name one of their nodes "Untitled" we should let them. They just can't have two of them. 😉